lib/Sema/TypeCheckAccess.cpp

```diff
@@ -956,7 +956,8 @@ void AccessControlChecker::check(Decl *D) {
           return false;
         Type ty = inherited.getType();
         if (ty->is<ProtocolCompositionType>())
-          ty = ty->getExistentialLayout().superclass;
+          if (auto superclass = ty->getExistentialLayout().superclass)
+            ty = superclass;
```

Comment:
There’s another copy of this check in the UsableFromInlineChecker too. I know that’s bad and I need to clean it up, but can you fix it and add a test?